pkg/cloudprovider/cloudprovider.go

```diff
@@ -100,30 +103,31 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
-func (c *CloudProvider) GetDriftedNodes(ctx context.Context, provisioner *v1alpha5.Provisioner, nodes []v1.Node) []v1.Node {
-	driftedNodesMap := map[string]bool{}
-	var driftedNodes []v1.Node
-	//Consider if provisioner should be a part of this interface
-	for _, drifter := range []interface {
-		GetDriftedNodes([]v1.Node) []v1.Node
-	}{
-		&AmiDrifter{
-			provisioner: provisioner,
-			ctx:         ctx,
-			c:           c,
-		},
-	} {
-		//Execute Drifters
-		for _, node := range drifter.GetDriftedNodes(nodes) {
-			//Check first in the map, one node can be drifted by multiple drifter implementations
-			if _, ok := driftedNodesMap[node.Name]; !ok {
-				driftedNodes = append(driftedNodes, node)
-			}
-			driftedNodesMap[node.Name] = true
+func (c *CloudProvider) IsNodeDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, node v1.Node) bool {
```

Comment:
I'm not convinced that this should be part of the cloud provider. Is there a reason we can't encapsulate all of this logic into the drift controller?

Reply:
I think the drift is tied to a cloudprovider in the sense that it needs information related to what cloudprovider has. Going forward, multiple drifters can be implemented in this function which need more access to cloudprovider. Ami is just one of the things which was rooted in cloudprovider -> instanceProvider -> launchTemplateProvider -> amiFamily.

Reply:
I am okay to convert it to `GetAmiForProvisioner` sorts, but I think we will have to expose more functions like this to get attributes related to cloudprovider going ahead.

Reply:
Sorry -- I don't have this fully clear in my head, so I may be giving bad advice. 

Previously, this cloudprovider.go file was a strict implementation of https://github.com/aws/karpenter-core/blob/main/pkg/cloudprovider/types.go#L47. I'd like to keep these 1:1 if at all possible. This raises the question of whether or not this drift controller implementation should be vendor neutral (e.g. karpenter core) or vendor specific.

I could see an argument that the drift logic could be standardized across vendors, with this as the extensibility point.

I see our options as:
1. Make drift an AWS specific controller for now (as you have it here), and promote it to be neutral when it makes sense
2. Make IsDrifted(v1.Node) part of the cloudprovider interface, and implement a vendor neutral controller that calls it

The more I think on this, the more I have a slight preference for #2. In either solution, I think that we should communicate this state via a node annotation as you have it today.

Reply:
Based on the slack conversation, we decided to move ahead with the approach that is implemented in this PR, i.e. having a function not on the cloudprovider interface. 

It can be later added to the interface when there is more clarity on Machine CRD. 
Resolving this conversation, please feel free to open :)